Telemetry/Telemetry.cpp

```diff
@@ -0,0 +1,173 @@
+            {
+                string status;
+                getStringParameter("status", status);
+
+                if (status != "STARTED" && status != "COMPLETE")
+                {
+                    LOGERR("Only the 'STARTED' or 'COMPLETE' status is allowed");
+                    returnResponse(false);
+                }
+
+                string reportProfile;
+                getStringParameter("reportProfile", reportProfile);
+
+                JsonArray profiles;
+                getProfiles(profiles);
+
+                JsonArray::Iterator it = profiles.Elements();
+                while(it.Next())
+                {
+                    std::string profile = it.Current().String();
+                    if (profile == reportProfile)
+                    {
+                        std::string rfc = RFC_REPORT_PROFILE_ENABLE;
+                        rfc += profile;
+                        rfc += "Report.Enable";
+
+                        WDMP_STATUS wdmpStatus = setRFCParameter(RFC_CALLERID, rfc.c_str(), status == "COMPLETE" ? "true" : "false", WDMP_BOOLEAN);
+                        if (WDMP_SUCCESS != wdmpStatus)
+                        {
+                            LOGERR("Failed to set %s: %d", rfc.c_str(), wdmpStatus);
+                            returnResponse(false);
+                        }
+
+                        returnResponse(true);
+                    }
+                }
+            }
+            else
+            {
+                LOGERR("No 'status' parameter");
+                returnResponse(false);
+            }
+
+            returnResponse(true);
+        }
+
+        uint32_t Telemetry::logApplicationEvent(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+
+            if (parameters.HasLabel("eventName") && parameters.HasLabel("eventValue"))
+            {
+                string eventName;
+                getStringParameter("eventName", eventName);
+
+                string eventValue;
+                getStringParameter("eventValue", eventValue);
+
+                LOGT2((char *)eventName.c_str(), (char *)eventValue.c_str());
+            }
+            else
+            {
+                LOGERR("No 'eventName' or 'eventValue' parameter");
+                returnResponse(false);
+            }
+
+            returnResponse(true);
+        }
+
+        void Telemetry::getProfiles(JsonArray &profiles)
+        {
+            profiles.Clear();
+
+            DIR *d = opendir("/opt/.t2persistentfolder/");
+
+            if (NULL != d)
+            {
+                struct dirent *de;
+
+                while ((de = readdir(d)))
+                {
+                    if (0 == de->d_name[0])
+                        continue;
+
+                    if (0 == strcmp(de->d_name, ".") || 0 == strcmp(de->d_name, ".."))
+                        continue;
+
+                    profiles.Add(de->d_name);
```

Comment:
Pls add json object here {"name":"profile_1"} instead of just names. We can then expand it in the future to add more information (key-value pairs) about the profiles if needed.

Reply:
Changed result of getAvailableReportProfiles(v1) to an array of objects, though Solution Overview in RDK-29174 says it should be array of strings.